Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-43.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
Sha: f6c0cd2ad602501524851d66dbdc47444b3a889d
Containerized:False
Number of Error(s): 13
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
\terrors: 0
/home/centos/smoketest/PR-15878/HPCC-Platform/roxie/ccd/ccdactivities.cpp:217:45: error: use of deleted function \xe2\x80\x98ActivityArray::ActivityArray(const ActivityArray&)\xe2\x80\x99
/home/centos/smoketest/PR-15878/HPCC-Platform/roxie/ccd/ccdquery.hpp:180:7: error: use of deleted function \xe2\x80\x98IArrayOf<IActivityFactory>::IArrayOf(const IArrayOf<IActivityFactory>&)\xe2\x80\x99
/home/centos/smoketest/PR-15878/HPCC-Platform/system/jlib/jlib.hpp:126:7: error: use of deleted function \xe2\x80\x98IArray::IArray(const IArray&)\xe2\x80\x99
/home/centos/smoketest/PR-15878/HPCC-Platform/system/jlib/jlib.hpp:45:17: error: use of deleted function \xe2\x80\x98OwnedReferenceArrayOf<IInterface>::OwnedReferenceArrayOf(const OwnedReferenceArrayOf<IInterface>&)\xe2\x80\x99
/home/centos/smoketest/PR-15878/HPCC-Platform/system/jlib/jarray.hpp:421:7: error: use of deleted function \xe2\x80\x98ArrayOf<IInterface*, IInterface&, OwnedReferenceArrayMapper<IInterface> >::ArrayOf(const ArrayOf<IInterface*, IInterface&, OwnedReferenceArrayMapper<IInterface> >&)\xe2\x80\x99
/home/centos/smoketest/PR-15878/HPCC-Platform/system/jlib/jarray.hpp:147:7: error: use of deleted function \xe2\x80\x98AllocatorOf<8>::AllocatorOf(const AllocatorOf<8>&)\xe2\x80\x99
/home/centos/smoketest/PR-15878/HPCC-Platform/system/jlib/jarray.hpp:102:32: error: use of deleted function \xe2\x80\x98Allocator::Allocator(const Allocator&)\xe2\x80\x99
make[2]: *** [roxie/ccd/CMakeFiles/ccd.dir/build.make:76: roxie/ccd/CMakeFiles/ccd.dir/ccdactivities.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:15459: roxie/ccd/CMakeFiles/ccd.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2022-03-09_16-32-33
time logs started
Prep time   :    17 sec (00:00:17)
Build time  :   272 sec (00:04:32)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   289 sec (00:04:49)
time logs end

[Build error:](https://gist.github.com/4152043acca8fee8f3c8d13f61994e16)
Build log [Build log](https://gist.github.com/HPCCSmoketest/4152043acca8fee8f3c8d13f61994e16#file-result-22-03-09-16-27-28-log)

--- a/esp/bindings/SOAP/Platform/soapparam.hpp
+++ b/esp/bindings/SOAP/Platform/soapparam.hpp
@@ -598,7 +598,7 @@ public:
     void copy(SoapArrayParam<arraytype,itemtype> &from)
     {
         arr.kill();
-        arraytype fromArray = from.getValue();
+        const arraytype & fromArray = from.getValue();
         ForEachItemIn(idx, fromArray)
           arr.append(fromArray.item(idx));
     }

--- a/esp/services/ws_sql/SQL2ECL/SQLExpression.cpp
+++ b/esp/services/ws_sql/SQL2ECL/SQLExpression.cpp
@@ -747,7 +747,8 @@ SQLFunctionExpression::SQLFunctionExpression(const char* funcname)
 SQLFunctionExpression::SQLFunctionExpression(const char* funcname, const IArrayOf<ISQLExpression> &params)
 {
     setFunction(funcname);
-    this->params = params;
+    ForEachItemIn(i, params)
+        this->params.append(OLINK(params.item(i)));
     this->alias = "";
     distinct = false;
 }

--- a/system/jlib/jarray.hpp
+++ b/system/jlib/jarray.hpp
@@ -66,6 +66,10 @@ typedef int (* StdCompare)(const void *_e1, const void *_e2);
 class jlib_decl Allocator
 {
 public:
+    Allocator() = default;
+    Allocator(const Allocator & ) = delete;
+    Allocator & operator = (const Allocator & ) = delete;
+
     void  kill();
     inline bool isItem(aindex_t pos = 0) const    { return pos < used; }      /* Is there an item at pos */
     inline aindex_t length() const                 { return used; } /* Return number of items  */

